testsrc/org/mozilla/javascript/tests/JavaAdapterInvokeTest.java

```diff
@@ -0,0 +1,111 @@
+package org.mozilla.javascript.tests;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.Scriptable;
+
+public class JavaAdapterInvokeTest {
+    Context cx = null;
+    Scriptable topScope = null;
+
+    @Before
+    public void enterContext() {
+        cx = Context.enter();
+        cx.setOptimizationLevel(-1);
```

Comment:
Dunno what our policy is, but should this test not also be run in non-interpreted mode?

Reply:
Usung `Utils.runWithAllOptimizationLevels` now